src/data/feature_index.js

```diff
@@ -269,6 +295,16 @@ register(
 
 export default FeatureIndex;
 
+function evaluateProperties(serializedProperties, styleLayerProperties, evaluationParameters) {
+    const evaluatedProperties = {};
+    for (const property in serializedProperties) {
+        const prop = styleLayerProperties instanceof PossiblyEvaluated ? styleLayerProperties.get(property) : null;
+        evaluatedProperties[property] = prop && prop.evaluate ? prop.evaluate(evaluationParameters.feature, evaluationParameters.featureState, evaluationParameters.availableImages) : prop;
+    }
+
+    return evaluatedProperties;
+}
```

Comment:
Nit: we have a `mapObject` function in `util`, so you could simplify this code to just the callback function for mapping.